services/hooks/src/api.js

```diff
@@ -368,24 +370,22 @@ builder.declare({
     });
   }
 
-  try {
-    await hook.modify((hook) => {
-      hook.metadata = hookDef.metadata;
-      hook.bindings = hookDef.bindings;
-      hook.task = hookDef.task;
-      hook.triggerSchema = hookDef.triggerSchema;
-      hook.schedule = schedule;
-      hook.nextTaskId = taskcluster.slugid();
-      hook.nextScheduledDate = nextDate(schedule);
-    });
-  } catch (err) {
-    if (err && err.code === 'PropertyTooLarge') {
-      return res.reportError('InputError', err.toString(), {});
-    }
-    throw err;
-  }
+  hook = hookUtils.fromDbRows(
+    await this.db.fns.update_hook(
+      hook.hookGroupId, /* hook_group_id */
+      hook.hookId, /* hook_id */
+      hookDef.metadata, /* metadata */
+      hookDef.task, /* task */
+      JSON.stringify(hookDef.bindings), /* bindings */
+      JSON.stringify(schedule), /* schedule */
```

Comment:
can we have the note here as well about json-arrays not being directly handled by node-pg?